packages/arb-token-bridge-ui/src/util/withdrawals/fetchWithdrawals.ts

```diff
@@ -124,9 +124,23 @@ export const fetchWithdrawals = async ({
     })
   ])
 
+  // get txs to be displayed on the current page (event logs)
+  const currentPageStart = pageNumber * pageSize
+  const currentPageEnd = currentPageStart + pageSize
+
+  const partialEthWithdrawalsFromEventLogs = [...ethWithdrawalsFromEventLogs]
+    // event logs start from the earliest, we need to reverse them
+    .reverse()
+    .slice(currentPageStart, currentPageEnd)
+  const partialTokenWithdrawalsFromEventLogs = [
+    ...tokenWithdrawalsFromEventLogs
+  ]
+    .reverse()
+    .slice(currentPageStart, currentPageEnd)
+
   const mappedTokenWithdrawalsFromEventLogs = (
     await Promise.all([
-      ...tokenWithdrawalsFromEventLogs.map(withdrawal =>
+      ...partialTokenWithdrawalsFromEventLogs.map(withdrawal =>
```

Comment:
Not introduced by this PR, but we don't need to spread here, we can use `Promise.all(partialTokenWithdrawalsFromEventLogs.map()...)`